packages/stock-location/src/services/stock-location.ts

```diff
@@ -204,35 +223,42 @@ export default class StockLocationService {
    * Updates an address for a stock location.
    * @param {string} addressId - The ID of the address to update.
    * @param {StockLocationAddressInput} address - The update data for the address.
-   * @param {Object} context - Context for the update.
-   * @param {EntityManager} context.manager - The entity manager to use for the update.
    * @returns {Promise<StockLocationAddress>} - The updated stock location address.
    */
 
   protected async updateAddress(
     addressId: string,
-    address: StockLocationAddressInput,
-    context: { manager?: EntityManager } = {}
+    address: StockLocationAddressInput
   ): Promise<StockLocationAddress> {
-    const manager = context.manager || this.getManager()
-    const locationAddressRepo = manager.getRepository(StockLocationAddress)
-
-    const existingAddress = await locationAddressRepo.findOne(addressId)
-    if (!existingAddress) {
+    if (!isDefined(addressId)) {
       throw new MedusaError(
         MedusaError.Types.NOT_FOUND,
-        `StockLocation address with id ${addressId} was not found`
+        `"addressId" must be defined`
       )
     }
 
-    const toSave = locationAddressRepo.merge(existingAddress, address)
+    return await this.atomicPhase_(async (manager) => {
+      const locationAddressRepo = manager.getRepository(StockLocationAddress)
 
-    const { metadata } = address
-    if (metadata) {
-      toSave.metadata = setMetadata(toSave, metadata)
-    }
+      const existingAddress = await locationAddressRepo.findOne({
+        id: addressId,
+      })
+      if (!existingAddress) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_FOUND,
+          `StockLocation address with id ${addressId} was not found`
+        )
+      }
+
+      const toSave = locationAddressRepo.merge(existingAddress, address)
+
+      const { metadata } = address
+      if (metadata) {
```

Comment:
question: Once merged, the metadata has already been merged as well isn't it?

Reply:
typeorm `merge` doesn't have the same behavior `setMetadata` has.

Reply:
I understand that, wanted you opinion on the global approach. Does it make sense to merge the metadata and the re copying them all. I would say that we can extract the metadata before the merge and then call the setMetada to avoid any side effects. Wdyt?

Reply:
fixed